edx_arch_experiments/kafka_consumer/management/commands/consume_events.py

```diff
@@ -103,17 +111,26 @@ def create_consumer(self, group_id):
 
         return DeserializingConsumer(consumer_config)
 
-    def handle_message(self, msg):
+    def emit_signals_from_message(self, msg):
         """
-        Place holder methods for how to handle an incoming message from the event bus
+        Determine the correct signal and send the event from the message
         """
-        # TODO (EventBus):
-        # Rewrite this to construct and/or emit the signal eventually specified in the message.
-        logger.info(f"Received message with key {msg.key()} and value {TrackingEvent.to_dict(msg.value())}")
+        if msg.headers():
+            event_types = [value for key, value in msg.headers() if key == EVENT_TYPE_HEADER]
+            if len(event_types) == 0:
+                # TODO (EventBus): iterate on error handling for missing signal type
+                logger.warning(f"Missing {EVENT_TYPE_HEADER} header on message, cannot determine signal")
+            # there really should only be one event type but theoretically we can have multiple
+            for event_type in event_types:
+                # TODO (EventBus): Figure out who is doing the encoding and get the
+                #  right one instead of just guessing utf-8
+                signal = OpenEdxPublicSignal.get_signal_by_type(event_type.decode("utf-8"))
```

Comment:
What does this raise if not found? Non-blocking, but seems like a useful quick except we could add with a log message.